examples/systemcalls/systemcalls.c

```diff
@@ -92,6 +144,39 @@ bool do_exec_redirect(const char *outputfile, int count, ...)
  *   The rest of the behaviour is same as do_exec()
  *   
 */
+    int pid;								//pid to store the return value of fork
+    int status;								//check the status for waitpid
+    int fd = open(outputfile, O_WRONLY|O_TRUNC|O_CREAT, 0644);		//open the outputfile for writinf the path in 
+    if (fd < 0){perror("open");exit(-1);}				//return error if could not open the file and exit with status -1
+
+    //fork
+    pid = fork();							
+    if(pid == -1){perror("fork"); exit(-1);}				//return error if child process not created and exit with status -1
+
+    else if(pid ==  0){							
```

Comment:
@Mich2899 think about how you could combine logic between this function and the one above.